api/python/quilt3/data_transfer.py

```diff
@@ -483,16 +502,17 @@ def _upload_or_copy_file(ctx, size, src_path, dest_bucket, dest_path):
             pass
         else:
             # Check the ETag.
-            dest_size = resp['ContentLength']
+            dest_size = resp['ObjectSize']
             dest_etag = resp['ETag']
             dest_version_id = resp.get('VersionId')
             if size == dest_size:
                 src_etag = _calculate_etag(src_path)
                 if src_etag == dest_etag:
                     # Nothing more to do. We should not attempt to copy the object because
                     # that would cause the "copy object to itself" error.
+                    sha256 = resp['Checksum'].get('ChecksumSHA256')
```

Comment:
It would be easier to read if we didn't call this `sha256` but `checksum`. Would also help to have a comment below for each case:
1) object in S3 and has checksum
2) object in S3 and does not have checksum
3) object local and we will request checksum
Or similar.